### PR TITLE
docs: add troubleshooting for root span error details

### LINKED ISSUE #1570: new django integration: uncaught view errors cause is no longer annotated on root span
With the introduction of the new django integration, any uncaught errors are no longer annotated at the root span.

Previously, when a view crashed, the corresponding error would be annotated at the `django.request` span and become searchable via analytics. Now, the only annotation is `error:true` and the actual error sitting on a `django.view` span further down in the stack.

One workaround that might be suggested is to index all `django.view` spans. This isn't useful as that span has no other annotations to break down results.

Old annotation:
old annotation

New annotation
new annotation
image

### PR DESCRIPTION
**Pull request by majorgreys**
Closes #1570

### CODE REVIEW

File: docs/troubleshooting.rst [hunk 1]
```diff
@@ -42,6 +42,45 @@ For more information about the ``version`` tag please see: https://docs.datadogh
...
+
+  class ErrorFilter(TraceFilter):
+    def process_trace(self, trace):
+        if not trace:
+            return trace
+
+        local_root = trace[0]
+
+        for span in trace[1:]:
+            if span.error == 1: # or any other conditional for finding the relevant child span
```
Comment (brettlangdon) [inline docs/troubleshooting.rst:69]:
```suggestion
            if span.error == 1:  # or any other conditional for finding the relevant child span
```

File: docs/troubleshooting.rst [hunk 2]
```diff
@@ -42,6 +42,45 @@ For more information about the ``version`` tag please see: https://docs.datadogh
...
+This can be a problem for users who want to see error details from a child span in the root span of a trace. An example of this is when an error is raised during the execution of view code, subsequently handled by the web framework in order to return an error page, the error details from the span for an instrumented view will not be included in the root span for the request.
+
+While this is default behavior for integrations, users can add a trace filter to propagate the error details up to the root span::
+
+  from ddtrace import Span, tracer
+  from ddtrace.filters import TraceFilter
+
+
+  class ErrorFilter(TraceFilter):
+    def process_trace(self, trace):
```
Comment (brettlangdon) [inline docs/troubleshooting.rst:62]: comment here describing the filter would be good.

e.g. find the first child span with an error and add it's error tags onto the root span.

**Review by brettlangdon — DISMISSED** [review]
nits, but otherwise lgtm

File: docs/troubleshooting.rst (hunk 2 above)
Comment (majorgreys) [inline docs/troubleshooting.rst:62]:
```suggestion
    def process_trace(self, trace):
        # Find first child span with an error and copy its error details to root span
```
@brettlangdon wdyt?

Comment (brettlangdon) [reply docs/troubleshooting.rst:62]: lgtm, think even just this basic comment is helpful to share the purpose

Comment (brettlangdon) [reply docs/troubleshooting.rst:62]: we need to assume someone will copy/paste this into their codebase without adding any context, so we should be sure the necessary context is there for them

**Review by brettlangdon — APPROVED** [review]

**Review by ZStriker19 — DISMISSED** [review]

**Review by brettlangdon — APPROVED** [review]

**Review by ZStriker19 — APPROVED** [review]

**Pull request merged**